src/services/factories/tokenList.ts

```diff
@@ -2,15 +2,26 @@ import { TokenList } from 'api/tokenList/TokenListApi'
 import { SubscriptionCallback } from 'api/tokenList/Subscriptions'
 import { ExchangeApi } from 'api/exchange/ExchangeApi'
 import { TokenDetails, Command } from 'types'
-import { logDebug } from 'utils'
+import { logDebug, retry } from 'utils'
 
-export function getTokensFactory(factoryParams: {
-  tokenListApi: TokenList
-  exchangeApi: ExchangeApi
-}): (tokenId: number) => TokenDetails[] {
+import { TokenFromErc20Params, TokenFromErc20 } from './'
+
+interface Injects {
+  getTokenFromErc20(params: TokenFromErc20Params): Promise<TokenFromErc20>
+}
+
+export function getTokensFactory(
+  factoryParams: {
+    tokenListApi: TokenList
+    exchangeApi: ExchangeApi
+  },
+  injects: Injects,
+): (tokenId: number) => TokenDetails[] {
   const { tokenListApi, exchangeApi } = factoryParams
+  const { getTokenFromErc20 } = injects
 
-  const updatedIdsForNetwork = new Set<number>()
+  // Set containing ids for networks which we successfully updated the tokens from the contract
+  const areTokensUpdated = new Set<number>()
```

Comment:
nitty: `areBlahBlah` makes me auto assume `boolean` - why not `updatedTokenSet`? or other. Anyways, doesn't really matter

Reply:
nyeh leave it